internal/highway/x/ipfs/protocol.go

```diff
@@ -4,39 +4,42 @@ import (
 	"bytes"
 	"context"
 	"fmt"
-	"io/ioutil"
-
 	ipfslite "github.com/hsanjuan/ipfs-lite"
 	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
 	"github.com/ipld/go-ipld-prime/codec/dagjson"
 	"github.com/ipld/go-ipld-prime/datamodel"
 	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+	"github.com/sonr-io/sonr/internal/highway/x/store"
 	"github.com/sonr-io/sonr/pkg/host"
 	st "github.com/sonr-io/sonr/x/schema/types"
+	"io/ioutil"
+	"log"
 )
 
-// IPFSProtocol leverages the IPFSLite library to provide simple file operations.
-type IPFSProtocol struct {
-	ctx       context.Context
-	node      host.SonrHost
-	dataStore *MemoryStore
+// Protocol leverages the IPFSLite library to provide simple file operations.
+type Protocol struct {
+	ctx   context.Context
+	host  host.SonrHost
+	store *store.Store
 	*ipfslite.Peer
 }
 
-// New creates a new IPFSProtocol instance with Host Implementation
-func New(ctx context.Context, host host.SonrHost) (*IPFSProtocol, error) {
+// New creates a new Protocol instance with Host Implementation
+func New(ctx context.Context, host host.SonrHost) (*Protocol, error) {
+	ds := store.New("tmp")
```

Comment:
Use `os.TempDir()` instead of "tmp" to support more platforms.

Reply:
I was actually aiming for a root path's `tmp` dir and not the system one's, but system tmp seems better indeed.